WeakAuras/Prototypes.lua

```diff
@@ -2373,67 +2373,87 @@ Private.event_prototypes = {
+          if repInfo and repInfo.friendshipFactionID > 0 then
+            standing = repInfo.reaction
+            if repInfo.nextThreshold then
+              minValue, maxValue, currentValue = repInfo.reactionThreshold, repInfo.nextThreshold, repInfo.standing
+            else
+              -- max rank, make it look like a full bar
+              minValue, maxValue, currentValue = 0, 1, 1
+              isCapped = true
             end
+            local rankInfo = C_GossipInfo.GetFriendshipReputationRanks(factionID)
+            if rankInfo.maxLevel > 0 then
+              friendshipRank, friendshipMaxRank = rankInfo.currentLevel, rankInfo.maxLevel
+            else
+              friendshipRank, friendshipMaxRank = 1, 1
+            end
+          end
 
-            if C_Reputation.IsFactionParagon(factionID) then
-              local currentValue, threshold, rewardQuestID, hasRewardPending, tooLowLevelForParagon = C_Reputation.GetFactionParagonInfo(factionID)
-              if currentValue then
-                bottomValue, topValue = 0, threshold
-                earnedValue = currentValue %% threshold
-                if hasRewardPending then
-                  earnedValue = earnedValue + threshold
-                end
-              else
-                currentValue, bottomValue, topValue, earnedValue = 0, 0, 0, 0
+          -- check if this is a Major faction (renown)
+          local renownLevel, maxRenownLevel, isWeeklyRenownCapped
+          local isMajorFaction = factionID and C_Reputation.IsMajorFaction(factionID)
+          if isMajorFaction then
+            local majorFactionData = C_MajorFactions.GetMajorFactionData(factionID)
+            minValue, maxValue = 0, majorFactionData.renownLevelThreshold
+            isCapped = C_MajorFactions.HasMaximumRenown(factionID)
+		        currentValue = isCapped and majorFactionData.renownLevelThreshold or majorFactionData.renownReputationEarned or 0
+            standing = RENOWN_LEVEL_LABEL .. majorFactionData.renownLevel
+            renownLevel = majorFactionData.renownLevel
+            local renownLevels = C_MajorFactions.GetRenownLevels(factionID)
+            maxRenownLevel = renownLevels and renownLevels[#renownLevels].level
+            isWeeklyRenownCapped = C_MajorFactions.IsWeeklyRenownCapped(factionID)
+          end
+
+          -- check if this is a faction with a paragon track
+          local isParagon = factionID and C_Reputation.IsFactionParagon(factionID)
+          if isParagon then
+            local paragonCurrentValue, paragonBarMax, rewardQuestID, hasRewardPending, tooLowLevelForParagon = C_Reputation.GetFactionParagonInfo(factionID)
+            if paragonCurrentValue then
+              minValue, maxValue = 0, paragonBarMax
+              currentValue = paragonCurrentValue %% paragonBarMax
+              if hasRewardPending then
+                currentValue = currentValue + paragonBarMax
               end
+              isCapped = currentValue >= maxValue
```

Comment:
This is different from before, previously capped was simply HasMaximumRenown()

Reply:
Ah storing isCapped is new, so that's not acutally new.

Anyway, what made you decide for this behaviour for capped?

Reply:
I'm not sure if you mean HasMaximumRenown, I think you mean hasRewardPending? Since you're referring to the paragon block. 
`isCapped = currentValue >= maxValue`  is the same as setting `isCapped = hasRewardPending`, though it makes more sense to use hasRewardPending.

Reply:
The point is what does "is capped" mean if you have a paragon reputation?

I think there are multiple ways to treat that:
* is capped == exalted || max renown, so ignoring paragon levels for is capped
* is capped == can't earn additional reputation, so false for paragon reputations
* is capped == hasRewardPending

Reply:
@InfusOnWoW my idea was is capped == hasRewardPending
But do you propose I split capped and hasRewardPending into seperate options? In that case, capped for a paragon rep would always be true, since you are max rep/renown/friendship.

Reply:
I frankly have no firm opinion on what is best. Someone that cares deeply about reputations should decide that. As in, you got to decide that.

And yes one option is to use different names

The things you have to consider are:
* How does the api behave. In general we keep close to what the apis view of the world is. As that's the most likely to keep working when blizzard introduces new concepts.
* What does the user expect when they read "is capped".
* What's the most useful, what are the use cases where a user would enable the "is capped" check.

Reply:
Paragon is quite a special case :\\
What I've done now:
* Consider `isCapped = true` when player is max exalted/friendship rank/renown level.
* If there is a paragon and player is in paragon: `isCapped = true` following the rule above.
* Paragon is an endless reputation and repeats itself every 10000 points. Previously we would display a player with a paragon reward pending and 74 points as `current: 10074` `total: 10000`. 
* This is now changed to `current: 74` `total: 10000`
* Added a Paragon Reward Pending option.